File: 230676-Products-Service/GESINV.ProductsService/GESINV.ProductsService.Dominio/Producto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GESINV.ProductsService.Dominio
{
    public class Producto
    {
        public Guid Id { get; set; }
        public string Nombre { get; set; }
        public string Descripcion { get; set; }
        public string ImagenPath { get; set; }
        public decimal Precio { get; set; }
        public int CantidadEnInventarioInicial { get; set; }
        public bool Accesible { get; set; }
        public int CantidadVendida { get; set; }
        public int CantidadComprada { get; set; }
        public Guid EmpresaId { get; set; }
        public List<DetalleCompraProducto> DetallesComprasProductos { get; set; }
        public List<DetalleVentaProducto> DetallesVentasProductos { get; set; }

        public int CantidadEnInventario()
        {
            return CantidadEnInventarioInicial + CantidadComprada - CantidadVendida;
        }
    }
}

File: 230676-API-Gateway-Service/GESINV.ApiGateway/modelsout/producto.go
package modelsout

type Producto struct {
	Id                          string  `json:"id"`
	Nombre                      string  `json:"nombre"`
	Descripcion                 string  `json:"descripcion"`
	ImagenPath                  string  `json:"imagenPath"`
	Precio                      float64 `json:"precio"`
	CantidadEnInventarioInicial int     `json:"cantidadEnInventarioInicial"`
	CantidadComprada            int     `json:"cantidadComprada"`
	CantidadVendida             int     `json:"cantidadVendida"`
	CantidadEnInventario        int     `json:"cantidadEnInventario"`
}
